rust/src/domains/polars/array/mod.rs

```diff
@@ -0,0 +1,53 @@
+use std::sync::Arc;
+
+use polars::series::Series;
+
+use crate::{core::Domain, error::Fallible};
+
+use super::DynSeriesElementDomain;
+
+#[cfg(feature = "ffi")]
+mod ffi;
+
+/// A domain that represents enum data.
+///
+/// Enum data is ostensibly a string,
+/// however the data is stored as a vector of indices into an encoding.
+/// This gives memory speedups when the number of unique values is small.
+///
+/// Differs from the ArrayDomain in that the categories are fixed and known.
+#[derive(Clone, Debug)]
+pub struct ArrayDomain {
```

Comment:
Did you mean to change the name of the struct to something like `EnumDomain`? If this is the right name, maybe more explanation about how this `ArrayDomain` differs from some other `ArrayDomain`?

Reply:
Updated docs. These docs should talk about ArrayDomain.